nemo/collections/common/data/lhotse/cutset.py

```diff
@@ -30,38 +30,93 @@
+from nemo.collections.common.data.lhotse.text_adapters import (
+    LhotseTextAdapter,
+    LhotseTextPairAdapter,
+    NeMoMultimodalConversationJsonlAdapter,
+    NeMoSFTJsonlAdapter,
+)
 from nemo.collections.common.parts.preprocessing.manifest import get_full_path
 
 
-def read_cutset_from_config(config: DictConfig) -> Tuple[CutSet, bool]:
+def read_cutset_from_config(config: DictConfig | dict) -> Tuple[CutSet, bool]:
     """
     Reads NeMo configuration and creates a CutSet either from Lhotse or NeMo manifests.
 
     Returns a tuple of ``CutSet`` and a boolean indicating whether the data is tarred (True) or not (False).
     """
     # First, check if the dataset is specified in the new configuration format and use it if possible.
+    if not isinstance(config, DictConfig):
+        config = DictConfig(config)
     if config.get("input_cfg") is not None:
         return read_dataset_config(config)
     # Now, we'll figure out if we should read Lhotse manifest or NeMo manifest.
     use_nemo_manifest = all(config.get(opt) is None for opt in ("cuts_path", "shar_path"))
     if use_nemo_manifest:
-        assert (
-            config.get("manifest_filepath") is not None
-        ), "You must specify either: manifest_filepath, cuts_path, or shar_path"
-        is_tarred = config.get("tarred_audio_filepaths") is not None
-    else:
-        is_tarred = config.get("shar_path") is not None
-    if use_nemo_manifest:
-        # Read NeMo manifest -- use the right wrapper depending on tarred/non-tarred.
-        cuts = read_nemo_manifest(config, is_tarred)
+        if config.get("manifest_filepath") is None:
+            raise IncompleteConfigError("You must specify either: manifest_filepath, cuts_path, or shar_path")
+        cuts, is_tarred = read_nemo_manifest(config)
     else:
-        # Read Lhotse manifest (again handle both tarred(shar)/non-tarred).
-        cuts = read_lhotse_manifest(config, is_tarred)
+        cuts, is_tarred = read_lhotse_manifest(config)
     return cuts, is_tarred
 
 
-KNOWN_DATASET_CONFIG_TYPES = frozenset(("nemo", "nemo_tarred", "lhotse", "lhotse_shar", "txt", "txt_pair", "group"))
+class IncompleteConfigError(RuntimeError):
+    pass
+
+
+KNOWN_DATA_CONFIG_TYPES = {}
+
+
+def get_known_config_data_types() -> KeysView[str]:
+    """
+    Return the names of all registered data type parsers.
+
+    Example:
+
+        >>> get_known_config_data_types()
+        ["nemo", "nemo_tarred", "lhotse", ...]
+    """
+    return KNOWN_DATA_CONFIG_TYPES.keys()
+
+
+def get_parser_fn(data_type_name: str):
+    """
+    Return the parsing function for a given data type name.
+    Parsing function reads a dataloading config and returns a tuple
+    of lhotse ``CutSet`` and boolean indicating whether we should use
+    iterable dataset (True) or map dataset (False) mechanism ("is tarred").
+    """
+    return KNOWN_DATA_CONFIG_TYPES[data_type_name]
+
+
+def data_type_parser(name: str | list[str]):
```

Comment:
"Data type" because it is used to read serialized data in some format and parse it into "data types" introduced in this PR. Registry - yes, cutset - not sure... data type reader maybe? to me both sound equally good / bad.